packages/ui-date-time-input/src/DateTimeInput/index.tsx

```diff
@@ -134,6 +135,23 @@ class DateTimeInput extends Component<DateTimeInputProps, DateTimeInputState> {
             .month(this.state.iso.month())
             .year(this.state.iso.year())
         }
+        if (this.props.initialTimeForNewDate && !this.state.timeSelectValue) {
+          const hour = Number(this.props.initialTimeForNewDate.slice(0, 2))
+          const minute = Number(this.props.initialTimeForNewDate.slice(3, 5))
+          if (isNaN(hour) || isNaN(minute)) {
+            error(
+              false,
+              `[DateTimeInput] initialTimeForNewDate prop is not in the correct format. Please use HH:MM format.`
+            )
+          } else if (hour > 23 || minute > 59) {
```

Comment:
to be thorough I'd also add here a check for negative numbers. I'm sure that moment's parse-all parser would interpret them which is not good for us